common/djangoapps/util/storage.py

```diff
@@ -1,46 +1,49 @@
 """ Utility functions related to django storages """
 
-from typing import Optional
+from typing import Optional, List
 from django.conf import settings
-from django.core.files.storage import default_storage, storages
+from django.core.files.storage import storages
 from django.utils.module_loading import import_string
 
 
-def resolve_storage_backend(storage_key: str, legacy_setting_key: str, options: Optional[dict] = None):
+def resolve_storage_backend(
+        storage_key: str,
+        legacy_setting_key: str,
+        legacy_sec_setting_keys: List[str] = None,
+        options: Optional[dict] = None):
     """
     Configures and returns a Django `Storage` instance, compatible with both Django 4 and Django 5.
     Params:
         storage_key = The key name saved in Django storages settings.
         legacy_setting_key = The key name saved in Django settings.
+        legacy_sec_setting_keys = List of keys to get the storage class.
+            For legacy dict settings like settings.BLOCK_STRUCTURES_SETTINGS.get('STORAGE_CLASS'),
+            it is necessary to access a second-level key or above to retrieve the class path.
+        legacy_options = Kwargs for the storage class.
         options = Kwargs for the storage class.
     Returns:
         An instance of the configured storage backend.
     Raises:
         ImportError: If the specified storage class cannot be imported.
-
     Main goal:
         Deprecate the use of `django.core.files.storage.get_storage_class`.
-
     How:
         Replace `get_storage_class` with direct configuration logic,
         ensuring backward compatibility with both Django 4 and Django 5 storage settings.
     """
 
     storage_path = getattr(settings, legacy_setting_key, None)
+    if isinstance(storage_path, dict) and legacy_sec_setting_keys:
+        for deep_setting_key in legacy_sec_setting_keys:
+            # For legacy dict settings like settings.CUSTOM_STORAGE = {"BACKEND": "cms.custom.."}
+            storage_path = storage_path.get(deep_setting_key)
```

Comment:
If the legacy_sec_setting_keys has 2 items in the list, and the first one returns `None` I think this will throw an exception when you try `None.get...`

We should add a test case for that and I think it's fine to raise the exception but maybe catch and re-raise it with extra context.